modules/andes-core/broker/src/main/java/org/wso2/andes/server/queue/QueueEntryImpl.java

```diff
@@ -159,6 +166,12 @@ public boolean isAvailable()
         return _state == AVAILABLE_STATE;
     }
 
+    public boolean isTimelyDisposable() {
```

Comment:
you will do a map iteration with this (reading the config) in a performance critical path. I suggest lets cache this value somewhere and use it ( given that config value doesn't change during the runtime)